### PR TITLE
Remove finalizers from backups before syncing

### PR DESCRIPTION
**Pull request by nrb**
Signed-off-by: Nolan Brubaker <nolan@heptio.com>

### CODE REVIEW

**Comment by skriss** [comment]
Code lgtm. Please add unit test(s).

**Comment by skriss** [comment]
(lmk if you need any direction on the units!)

File: pkg/controller/backup_sync_controller_test.go [hunk 1]
```diff
@@ -77,6 +90,7 @@ func TestBackupSyncControllerRun(t *testing.T) {
 
 			// we only expect creates for items within the target bucket
 			for _, cloudBackup := range test.cloudBackups {
+				cloudBackup.Finalizers = stringslice.Except(cloudBackup.Finalizers, gcFinalizer)
```
Comment (skriss) [inline pkg/controller/backup_sync_controller_test.go]: I'd add a comment here just stating that we expect the controller to remove the gc finalizer.

It feels slightly weird that the code here to prep the expectation is identical to the code in the test, but I think given that we're separately testing `stringslice.Except`, what we're verifying here is just that it's actually called, so I think this is fine as-is.

File: pkg/util/test/test_backup.go [hunk 2]
```diff
@@ -124,3 +124,9 @@ func (b *TestBackup) WithResourceVersion(version string) *TestBackup {
 	b.ResourceVersion = version
 	return b
 }
+
+func (b *TestBackup) WithFinalizer(finalizers ...string) *TestBackup {
```
Comment (skriss) [inline pkg/util/test/test_backup.go]: nit: s/WithFinalizer/WithFinalizers/

File: pkg/controller/backup_sync_controller_test.go (hunk 1 above)
Comment (nrb) [reply pkg/controller/backup_sync_controller_test.go]: > It feels slightly weird that the code here to prep the expectation is identical to the code in the test, but I think given that we're separately testing stringslice.Except, what we're verifying here is just that it's actually called, so I think this is fine as-is.

Right, I agree. An alternative may be to call `!stringslice.Has(cloudBackup.Finalizers, gcFinalizer)` on each backup, in addition to verifying the backup passed in to the create action. WDYT?

Comment (skriss) [reply pkg/controller/backup_sync_controller_test.go]: I think it's ok as-is. This test fails if the wrong finalizer is removed, or if no finalizer is removed, which is what we want to see.

Comment (nrb) [reply pkg/controller/backup_sync_controller_test.go]: Actually, looking at this more, we're doing the exact same thing the code inside the `c.run()` call is doing, right? We're not even verifying it, really. So I think doing the `Has` check makes more sense.

**Pull request merged**